src/common/tests/SConscript

```diff
@@ -56,30 +54,34 @@ def scons():
     tenv.Install('$PREFIX/bin/', [common_test])
 
     unit_env = tenv.Clone()
-    # Some syscalls are aliased when fortification is enabled, which makes
-    # mocking in unit tests into a headache.
-    # The compiler decides this on its own, but let's force the issue.
-    compiler_setup.remove_fortify(unit_env)
+    unit_env.AppendUnique(OBJPREFIX='utest_')
+
+    unit_env.AppendUnique(LINKFLAGS=["-Wl,--wrap=socket", "-Wl,--wrap=connect", "-Wl,--wrap=bind",
```

Comment:
maybe this could be something like

functions=['socket', 'connect', ...]
for func in functions:
    unit_env.AppendUnique(LINKFLAGS=["-Wl,--wrap=%s" % func])

One might also consider a linker script.   See install/share/daos/ioil-ld-opts as an example

Reply:
Updated to use the linker script. Thanks for the suggestion!